staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -556,23 +571,43 @@ func (g *genValidations) emitValidationForType(c *generator.Context, inType *typ
 			if t.Kind == types.Struct || t.Kind == types.Alias {
 				// If this field is another type, call its validation function.
 				// Checking for nil is handled inside this call.
-				g.emitCallToOtherTypeFunc(c, t, childVarName, childPath, childIsPtr, sw)
+				g.emitCallToOtherTypeFunc(c, t, childIsPtr, bufsw)
 			} else {
 				// Descend into this field.
-				g.emitValidationForType(c, t, childVarName, childPath, sw, child.eachKey, child.eachVal)
+				g.emitValidationForType(c, t, childIsPtr, bufsw, child.eachKey, child.eachVal)
+			}
+
+			if buf.Len() > 0 {
+				if didSome {
+					sw.Do("\n", nil)
+				}
+				sw.Do("// field $.inType|raw$.$.fieldName$\n", targs)
+				//TODO: pass val first or fldpath first?  validators do fldpath, why?
+				sw.Do("errs = append(errs,\n", targs)
```

Comment:
> validators do fldpath, why?

No principled reason. We could change parameter orders to validators if needed.  (I think I was looking at some hand written validation code that put field path as the 1st argument when wrote the first validators, but AFAIK, the hand written code does not have a strong convention)

Reply:
Let's do this as a followup - decie and make it consistent.  My (slight) pref is for fldPath being after obj.